tests/Http/Resources/JsonApi/JsonApiResourceTest.php

```diff
@@ -37,4 +61,54 @@ public function testUnableToUnsetWrapper()
 
         JsonApiResource::withoutWrapping();
     }
+
+    public function testResourceTypeIsPickedFromMorph()
+    {
+        $model = new JsonApiModel(['id' => 1, 'name' => 'User']);
+
+        $responseData = $this->fakeJsonApiResponseForModel($model)['data'];
+
+        $this->assertArrayHasKey('type', $responseData);
+        $this->assertSame('json-api-models', $responseData['type']);
+    }
+
+    public function testIncludedResourceDoesNotContainPrimaryKey()
+    {
+        $model = new JsonApiModel(['id' => 1, 'name' => 'User']);
+        $model->setRelation('manager', new JsonApiModel(['id' => 2, 'name' => 'Manager']));
+        $model->setRelation('deputy', new JsonApiModel(['id' => 2, 'email' => 'deputy@example.com']));
+
+        $responseData = $this->fakeJsonApiResponseForModel($model);
+        $this->assertArrayNotHasKey('id', $responseData['included'][0]['attributes']);
+    }
+
+    public function testIncludedMatchingResourceAttributesAreMerged()
+    {
+        $model = new JsonApiModel(['id' => 1, 'name' => 'User']);
+        $model->setRelation('manager', new JsonApiModel(['id' => 2, 'name' => 'Manager']));
+        $model->setRelation('deputy', new JsonApiModel(['id' => 2, 'email' => 'deputy@example.com']));
+
+        $responseData = $this->fakeJsonApiResponseForModel($model);
+
+        $this->assertEquals([
+            'id' => '2',
+            'type' => 'json-api-models',
+            'attributes' => [
+                'name' => 'Manager',
+                'email' => 'deputy@example.com',
+            ],
+        ], $responseData['included'][0]);
+    }
```

Comment:
@timacdonald need your input on this and how it should be affected, especially when the resource (e.g., `UserResource`) needs `id`, `name`, and `email` values to be available, but the relationship only loads partial values.